app/controllers/answers_controller.rb

```diff
@@ -0,0 +1,86 @@
+class AnswersController < ApplicationController
+  include ApplicationHelper
+
+  def create
+    unless logged_in?
```

Comment:
This block could be a method that is then called as a before_action / before_filter